internal/provisioner/kops_provisioner.go

```diff
@@ -26,15 +26,37 @@ type KopsProvisioner struct {
 	s3StateStore      string
 	certificateSslARN string
 	logger            log.FieldLogger
+	aws               aws
+	privateDNS        string
 }
 
+// HelmDeployment deploys Helm charts.
+type helmDeployment struct {
+	valuesPath          string
+	chartName           string
+	namespace           string
+	chartDeploymentName string
+	dns                 string
+}
+
+// aws abstracts the aws client operations required by the installation supervisor.
+type aws interface {
+	CreateCNAME(dnsName string, dnsEndpoints []string, logger log.FieldLogger) error
+	DeleteCNAME(dnsName string, logger log.FieldLogger) error
+}
+
+// Array of helm apps that need DNS registration
+var helmApps = []string{"prometheus"}
+
 // NewKopsProvisioner creates a new KopsProvisioner.
-func NewKopsProvisioner(clusterRootDir, s3StateStore, certificateSslARN string, logger log.FieldLogger) *KopsProvisioner {
+func NewKopsProvisioner(clusterRootDir string, s3StateStore string, certificateSslARN string, logger log.FieldLogger, aws aws, privateDNS string) *KopsProvisioner {
```

Comment:
```suggestion
func NewKopsProvisioner(clusterRootDir, s3StateStore, certificateSslARN string, logger log.FieldLogger, aws aws, privateDNS string) *KopsProvisioner {
```

nit: you don't need to specify string for each variable when passing in multiple strings in a row

Reply:
Automatically added by linter, fixed.